src/store/settingsSlice.ts

```diff
@@ -60,6 +63,9 @@ const initialState: SettingsState = {
       accessToken: '',
     },
   },
+  signing: {
+    useOnChainSigning: false,
+  },
```

Comment:
I like the idea. But it would be a bigger change. I'd say lets see if users will run into the problem that specific Safe apps only support on-chain signing and then add it if needed.